source/develop/release-management/features/storage/cd-block-storage.md

```diff
@@ -0,0 +1,228 @@
+    1. fail with original error
+1. tear down drivespec
+  - if tearing down volume fails: log error (volume used by another VM)
+1. remove drivespec from VM metadata
+  - if failed: log error and succeed
+
+
+#### VM recovery flow
+
+Try to cleanup storage leftovers from insert and remove operations without changing VM state.
+
+1. metadata in progress, volume is not active
+  - remove metadata from XML
+1. metadata in progress, volume active, no device
+  - deactivate the volume
+  - remove metadata from XML
+1. metadata in progress, volume active, device exists
+  - remove attribute `in_progress` from metadata
+
+
+#### VM with loaded CD, changeCD
+
+1. engine sends ejectCD, specifying ISO path by PDIV format
+1. engine sends insetrCD command, specifying ISO by PDIV format
+
+##### Changes in Engine
+
+For engine only new internal commands will be added and there will be no changes visible to the user.
+
+- Update ChangeCD to use InsertCD and EjectCD on 4.5 cluster version
+
+On engine side also two new internal commands will be added:
+
+- Add InsertCD sub-command
+- Add EjectCD sub-command
+
+
+## Installation/Upgrade
+
+New vdsm APIs will we delivered with new vdsm packages contaning these changes.
+Changes in engine will require approprate vdsm version and use new vdsm APIs.
+Once user upgrades to this engine version, changes will be available to the user.
+
+
+## Testing
+
+### Test setup
+
+- create block storage domain
+- upload an ISO into block storage domain
+- create VM with CD
+
+### Preposed testing scenarios
+
+- eject CD from CD-ROM
+  - verify CD was removed in engine UI
+  - verify that metadata related to this CD-ROM was removed from VM XML
+  - verify ISO volume was deactivated on the host
+
+- load CD into CD-ROM
+  - verify CD was loaded in engine UI
+  - verify that metadata related to this CD-ROM was added into VM XML
+  - verify ISO volume was activated on the host
+  - verify ISO is not visible to VM (e.g by running `virsh domblklist`)
+
+- eject CD from CD-ROM
+  - verify CD was removed in engine UI
+  - verify that metadata related to this CD-ROM was removed from VM XML
+  - verify ISO volume was deactivated on the host
+  - verify ISO is visible to VM (e.g by running `virsh domblklist`)
+
+- change CD in CD-ROM
+  - verify new CD was loaded in engine UI
+  - verify that metadata related to old CD-ROM was removed from VM XML
+  - verify that metadata related to new CD-ROM was added into VM XML
+  - verify old ISO volume was deactivated on the host
+  - verify new ISO volume was activated on the host
+  - verify old ISO is not visible to VM (e.g by running `virsh domblklist`)
+  - verify ISO is visible to VM (e.g by running `virsh domblklist`)
+
+
+### Related links
+
+- [Discussion in ovirt devel mailing list.](https://www.mail-archive.com/devel@ovirt.org/msg16082.html)
+- [Bug 1725915: Vdsm tries to tear down in-use volume of ISO in block storage domain](https://bugzilla.redhat.com/1725915)
+
+
```

Comment:
Please proofread (spell checking, trivial grammar errors) before submitting the final version.